test/integration/client/client_test.go

```diff
@@ -1342,3 +1347,59 @@ func TestExtractModifyApply_ForceOwnership(t *testing.T) {
 		t.Errorf("createMgrExtracted apply configuration did not match expected, got:\n%s\n", cmp.Diff(expectedCreateExtracted, createMgrExtracted))
 	}
 }
+
+func TestAPIServerTransportMetrics(t *testing.T) {
+	result := kubeapiservertesting.StartTestServerOrDie(t, nil, nil, framework.SharedEtcd())
+	defer result.TearDownFn()
+
+	c := clientset.NewForConfigOrDie(result.ClientConfig).RESTClient()
+
+	body, err := c.Get().AbsPath("/metrics").DoRaw(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// TODO: this can be much better if there is some library that parse prometheus metrics
+	// the existing one in "k8s.io/component-base/metrics/testutil" uses the global variable
+	// but we want to parse the ones returned by the endpoint to be sure the metrics are
+	// exposed correctly
```

Comment:
Could [ScrapeAndCompare](https://github.com/kubernetes/component-base/blob/v0.27.0/metrics/testutil/testutil.go#L70-L75) perhaps cover your needs?

Reply:
Sure. I just wanted to make sure this didn't solve the problem you mentioned, but I am totally fine with your current approach :+1:

Reply:
it was a good suggestion